src/txn/txn.c

```diff
@@ -716,7 +721,8 @@ __txn_commit_timestamps_assert(WT_SESSION_IMPL *session)
 			if (F_ISSET(txn, WT_TXN_PREPARE)) {
 				WT_RET(__wt_open_cursor(session,
 				    op->btree->dhandle->name, NULL,
```

Comment:
A potential optimisation could be to check if we've already got a cursor opened for that dhandle. In that case I think we can just reuse the cursor. But the bookkeeping logic is going to become more complex.

Does anyone think this is a good idea?

Reply:
This is a rarely used debug path. I would do what is easiest and most readable.

Reply:
Understood. I'll keep it as is for the time being.

Reply:
For the record. It wouldn't be OK to re-use a cursor for another op, if we need to keep the op in cache. The cursor needs to remain positioned on the page until it would be OK for the page to be free'd, and re-using the cursor might leave it positioned on another page.